src/conventions/h-group/clue-interpretation/connecting-cards.js

```diff
@@ -159,15 +160,90 @@ function find_unknown_connecting(game, giver, target, reacting, identity, connec
 		}
 		// Finessed card is delayed playable
 		else if (game.level >= LEVEL.INTERMEDIATE_FINESSES && state.play_stacks[finesse.suitIndex] + 1 === finesse.rank) {
-			// Could be duplicated in giver's hand - disallow hidden finesse
-			if (state.hands[giver].some(c => c.clued && game.players[giver].thoughts[c.order].inferred.has(identity))) {
+			// This may be a bluff. resolve_bluff will rule out a bluff if it doesn't
+			// look like it could lead to the following plays.
+			const bluff = game.level >= LEVEL.BLUFFS &&
+				connected.length == 1 &&
+				((giver + 1) % state.numPlayers) == reacting &&
+				state.hands[reacting].filter(c => game.players[reacting].unknown_plays.has(c.order)).length == 0;
+
+			// Could be duplicated in giver's hand - disallow hidden finesse unless it could be a bluff.
+			if (!bluff && state.hands[giver].some(c => c.clued && game.players[giver].thoughts[c.order].inferred.has(identity))) {
 				logger.warn(`disallowed hidden finesse on ${logCard(finesse)} ${finesse.order}, true ${logCard(identity)} could be duplicated in giver's hand`);
 				return;
 			}
 
-			return { type: 'finesse', reacting, card: finesse, hidden: true, identities: [finesse.raw()] };
+			return { type: 'finesse', reacting, card: finesse, hidden: !bluff, bluff, identities: [finesse.raw()] };
+		}
+	}
+}
+
+/**
+ * Determines whether a bluff connection is a valid bluff, and updates the connection accordingly.
+ * @param {Game} game
+ * @param {number} giver 				The player index that gave the clue. They cannot deduce unknown information about their own hand.
+ * @param {number} target 				The player index receiving the clue. They will not find self-prompts or self-finesses.
+ * @param {Connection[]} connections	The complete connections leading to the play of a card.
+ * @param {ActualCard[]} promised		The non-hidden play(s) which are promised.
+ * @returns {Connection[]}
+ */
+export function resolve_bluff(game, giver, target, connections, promised) {
+	if (connections.length == 0 || !connections[0].bluff)
+		return connections;
+	const { state } = game;
+	let bluff_possible = true;
+	const bluffCard = connections[0].card;
+	// Determine the next play if this is a bluff.
+	let next_play = connections.findIndex(connection => connection.card == promised[0]) + 1;
+	if (next_play == 0)
+		next_play++;
```

Comment:
Is this part necessary? It seems like if there are no connections that are promised, that's because the only connection found was a bluff, which means that `bluff_possible` can't be false and we can skip doing the next loop. (It seems like incrementing `next_play` actually skips the loop too, but it isn't obvious that this is the case.) Then `next_play` could be `const`, and we could simplify this into:
```js
const next_play = connections.findIndex(connection => connection.card == promised[0]) + 1;
let bluff_possible = next_play === 0 ||
    connections.every((conn, index) => index < next_play || (!conn.hidden && conn.type !== 'finesse'));
```
I'm not confident though, correct me if I'm wrong.

Reply:
In fact, it's just plain not possible to not find the first promised card. It is either the bluff card or some other card in connections. Fixed!